File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& arr) {
        if(arr.empty()){
            return "";
        }
        string res = "";
        sort(arr.begin() , arr.end());
        string first = arr[0];
        string last = arr[arr.size()-1];
        for(int i = 0 ;i<first.length() && i<last.length() ; i++){
            if(first[i] == last[i]){
                res += first[i];
            }else{
                break;
            }
        }
        return res;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        n = len(strs)
        strs.sort()
        first = strs[0]
        last = strs[n-1]
        i = 0
        while i<len(first) and i<len(last) and first[i] == last[i]:
            i+=1
        return first[:i]

        
        